### PR TITLE
Fix(rollback): improve SafeMode

### LINKED ISSUE #279: Safe mode causes 0 mutationscore when active
When the new safe mode (#257) is activated it seems to cause a 0 mutationscore.

@dupdob Maybe you could take a look at this yourself?

### PR DESCRIPTION
**Pull request by dupdob**
Fix #279 and #283

### CODE REVIEW

**Comment by dupdob** [comment]
Working on the test issue

**Comment by dupdob** [comment]
Well it turns out removing brackets is harder than I thought. Closing this PR and will keep on working

**Comment by dupdob** [comment]
Fixed the issue by ensuring RemoveIfStatement returns a single statement when possible and a syntaxblock otherwise. The latter situation should not happen with current version of Stryker, but was present as a test.

**Comment by Mobrockers** [comment]
Could you create an issue explaining the syntax kind you found which we do not yet properly support and attach the 'UnsupportedSyntax' test? That way you won't have to fix the missing syntax support in this PR and we don't lose the clear example you created either.

File: src/Stryker.Core/Stryker.Core/MutationTest/MutationTestProcess.cs [hunk 1]
```diff
@@ -120,8 +120,14 @@ public void Mutate(StrykerOptions options)
                         _logger.LogInformation("{0} mutants got status {1}", mutantsToSkip.Count(), MutantStatus.Skipped.ToString());
                     }
                 }
+                else
+                {
+                    _logger.LogError("Failed to restore the project to a buildable state. Please report the issue. Stryker can not proceed further");
```
Comment (richardwerkman) [inline src/Stryker.Core/Stryker.Core/MutationTest/MutationTestProcess.cs]: Can this be moved to the CompilingProcess class?

Comment (dupdob) [reply src/Stryker.Core/Stryker.Core/MutationTest/MutationTestProcess.cs]: I guess I can, but this sounds strange to me: it means that compilingprocess.Compile either succeed or throws.
What benefit do you see there?

Comment (dupdob) [reply src/Stryker.Core/Stryker.Core/MutationTest/MutationTestProcess.cs]: No

**Pull request merged**